pt_miniscreen/core/component.py

```diff
@@ -0,0 +1,223 @@
+import logging
+import threading
+from time import sleep, time
+from typing import Any, Dict
+from weakref import WeakMethod
+
+from PIL import Image
+
+from .utils import is_same_image
+
+logger = logging.getLogger(__name__)
+
+
+class Interval(threading.Timer):
```

Comment:
Yeah I can see why this needs more explanation, I'll add some comments. For now to answer some of the questions:
- it inherits from Timer so that it has the cancel behaviour and so the semantics match, this will help with documentation since we can refer to the Timer docs directly. Overriding run is what would need to be done if we inherited from Thread so that would only mean we need to implement the Timer API ourselves which has it's own issues.
- I'm not too suprised the finished event is not in the docs since it's not an external API, it's pretty normal to use these internals when inheriting though!
- For the WeakMethod I'll add a comment for why it's there 👍 
- I think it's fair to say it would be helpful to explain the motivation for taking account of the execution time. I'll add that too 😄

Reply:
I've added some comments 👍